packages/react/src/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,387 @@
+              left: `${position?.left ?? 0}px`
+            },
+            id: `${instanceId}-menu`,
+            'aria-labelledby': ariaLabel,
+            children: Children.map(child.props.children, item => {
+              if (isValidElement(item)) {
+                return cloneElement(item as ReactElement<ActionMenuItemProps>, {
+                  handler: handleItemSelection,
+                  type: selectionVariant
+                })
+              }
+              return item
+            })
+          })
+        }
+      }
+      return acc
+    }, {})
+
+    return (
+      <div
+        id={instanceId}
+        className={clsx(styles.ActionMenu, disabled && styles['ActionMenu--disabled'])}
+        data-testid={testId || testIds.root}
+      >
+        {SelectButton}
+        {showMenu ? SelectOverlay : null}
+      </div>
+    )
+  }
+)
+
+type ActionMenuButtonProps = PropsWithChildren<Ref<HTMLButtonElement>> & {
+  'aria-label': string
+  id?: string
+  ref?: React.RefObject<HTMLButtonElement>
+  className?: string
+  menuOpen?: boolean
+  disabled?: boolean
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
+  'data-testid'?: string
+}
+
+const ActionMenuButton = forwardRef<HTMLButtonElement, ActionMenuButtonProps>(
+  ({children, className, 'data-testid': testId, disabled, menuOpen, ...props}, ref) => {
+    return (
+      <Button
+        ref={ref}
+        className={clsx(styles.ActionMenu__button, className)}
+        hasArrow={false}
+        aria-haspopup="true"
+        aria-expanded={menuOpen ? 'true' : 'false'}
+        trailingVisual={ChevronDownIcon}
+        disabled={disabled}
+        data-testid={testId || testIds.button}
+        {...props}
+      >
+        <span className={styles['ActionMenu__button-text']}>{children}</span>
+      </Button>
+    )
+  }
+)
+
+type ActionMenuItemProps = {
+  value: string
+  handler?: (newValue: string) => void
+  selected?: boolean
+  type?: 'none' | 'single'
+  disabled?: boolean
+} & PropsWithChildren<React.HTMLProps<HTMLLIElement>>
+
+const roleTypeMap = {
+  none: 'menuitem',
+  single: 'menuitemradio'
+}
+
+const ActionMenuItem = ({
+  children,
+  className,
+  disabled,
+  handler,
+  selected,
+  type,
+  value,
+  ...props
+}: ActionMenuItemProps) => {
+  return (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
+    <li
+      className={clsx(
+        styles.ActionMenu__item,
+        type === 'single' && styles[`ActionMenu__item--selection-type-${type}`],
+        className
+      )}
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-element-to-interactive-role
+      role={roleTypeMap[type || 'single']}
+      aria-checked={type === 'none' ? undefined : selected ? 'true' : 'false'}
+      aria-disabled={disabled ? 'true' : 'false'}
+      onClick={handler && !disabled ? () => handler(value) : undefined}
+      tabIndex={disabled ? undefined : 0}
```

Comment:
When a menu item is disabled, it loses the ability to be focused. Could we allow the `aria-disabled` items to still be focusable like the other menu items? When using a screen reader the amount of items in the menu will be announced, and this will include the `aria-disabled` items. If we allow the items to retain focus, a user knows what exactly is disabled.

Reply:
Good to know 👍 I've added back the ability to focus on this.

I initially set this to not focus on disabled items given that it's not interactive. Could this be a little misleading now that you can focus on it but not select the item though?

Reply:
Good question! Here's a bit [more context on why they should be focusable](https://github.com/github/accessibility/blob/main/docs/office-hours-notes/2022-08-03-menu-disabled-items-segmented-control-tree-view.md#menus-with-disabled-items). Mainly we'd want to make them focusable because a screen reader will announce the amount of items in a menu including the disabled ones, so we want to make sure that they can reach all of them. 

Without the disabled item being focusable, a user would hear that a menu has 3 items, but can only find 2 of them as the third is disabled. We want to make sure each item is discoverable, even if they can't trigger them. The good thing is, the state of the item (disabled) will be announced to the user as we're using `aria-disabled` here.

Reply:
Great info, thanks Tyler 🙏. I guess the key here is aria-disabled being set. We'll need to make sure this has adequate coverage against regressions.